### PR TITLE
app: fix debug mode propagation

### LINKED ISSUE #34: app: cli --debug option not setting api app in debug mode
```console
$ python manage.py --debug shell
Python 3.4.3 (default, Apr  7 2015, 08:05:21)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
App: mysite [debug]
Instance: /Users/lnielsen/envs/invenio3test/bin/../var/mysite-instance
>>> app.debug
True
>>> app.wsgi_app.mounts['/api'].debug
False
>>> app.config['DEBUG']
True
>>> app.wsgi_app.mounts['/api'].config['DEBUG']
False
```

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* Fixes the propagation of debug mode in
  wsgi_factory productions. (closes #34)

Signed-off-by: Javier Delgado <javier.delgado.fernandez@cern.ch>

### CODE REVIEW

File: invenio_base/app.py [hunk 1]
```diff
@@ -106,6 +106,9 @@ def _create_app(**kwargs):
             modules=blueprints,
         )
 
+        if 'debug' in kwargs:
```
Comment (jirikuncar) [inline invenio_base/app.py]: WDYT about?
```python
debug = kwargs.get('debug')
if debug is not None:
    app.debug = debug
```

File: invenio_base/app.py [hunk 2]
```diff
@@ -133,9 +136,10 @@ def create_cli_app(info):
             info.create_app = None
             app = info.load_app()
         else:
-            app = create_app()
```
Comment (jirikuncar) [inline invenio_base/app.py]: `app = create_app(debug=info.debug)`

**Comment by jirikuncar** [comment]
@JavierDelgadoFernandez please rebase on top of #44.

**Comment by jirikuncar** [comment]
@JavierDelgadoFernandez 
```
...
* Fixes the propagation of debug mode inwsgi_factory productions.
  (closes #34)
...
```

File: invenio_base/app.py [hunk 3]
```diff
@@ -216,8 +218,8 @@ def base_app(import_name, instance_path=None, static_folder=None,
     try:
         if not os.path.exists(instance_path):
             os.makedirs(instance_path)
-    except Exception:  # pragma: no cover
-        pass
+    except Exception as ex:  # pragma: no cover
+        app.logger.exception(ex)
```
Comment (lnielsen) [inline invenio_base/app.py]: ``app.logger.exception()``

Comment (JavierDelgadoFernandez) [inline invenio_base/app.py]: But that function takes at least one argument (Two arguments at least, I am not counting self).

Comment (lnielsen) [inline invenio_base/app.py]: Ah yes, but it should be a string message. The ``logger.exception`` automatically grabs the exception:
``app.logger.exception("Failed to create instance path")``
https://docs.python.org/2/library/logging.html#logging.Logger.exception

**Comment by JavierDelgadoFernandez** [comment]
@lnielsen Done.

File: invenio_base/app.py [hunk 4]
```diff
@@ -216,8 +218,8 @@ def base_app(import_name, instance_path=None, static_folder=None,
     try:
         if not os.path.exists(instance_path):
             os.makedirs(instance_path)
-    except Exception:  # pragma: no cover
-        pass
+    except Exception as ex:  # pragma: no cover
```
Comment (jirikuncar) [inline invenio_base/app.py]: unused `ex` variable

**Pull request closed without merging**